Normalisation graphique et accessibilité

AmauryCarrade opened the pull request:
La PR #5835 avait beaucoup d'ambition. Trop, en réalité. Cette PR vient faire une partie du travail à part : la normalisation du CSS et l'amélioration de l'accessibilité visuelle du site (contrastes, notamment). Le reste est laissé à plus tard : cette PR ne change pas drastiquement le style graphique de ZdS.

_Ce qui suit est une reprise mise à jour des parties pertinentes de l'autre PR._

## Structuration de la logique du _front-end_

Améliorer le site graphiquement c'est bien ; s'assurer qu'il restera propre et cohérent à l'avenir, c'est mieux.

Actuellement, les styles sont conçus de façon un peu… bordélique, et pour cause : au delà des deux couleurs principales de ZdS (à savoir, `hsl(199, 85%, 21%)` et `hsl(37, 93%, 58%)`), il n'y a absolument aucune norme graphique étayée. Bien sûr, il y a des conventions implicites, des composants réutilisés, et heureusement — mais rien de très cohérent.

L'objectif de cette PR est donc d'ajouter une forme de norme documentée concernant : 
- [x] les couleurs du site (**[voir la palette proposée](https://zestedesavoir.github.io/normes-graphiques/#couleurs)** ‑ à noter qu'elle n'est pas actuellement figée) ;
- [x] les tailles de police ;
- [x] les ombres ;
- [x] les arrondis de bordure ;
- [x] les longueurs en général (marges, bordures, …).

_Tous ces éléments sont visualisables sur [le site “Compagnon d'interface” développé rapidement pour l'occasion](https://zestedesavoir.github.io/normes-graphiques/)._

Cela peut sembler bureaucratique et potentiellement super casse pieds à utiliser derrière pour faire évoluer notre interface et créer de nouveaux composants graphiques, mais en réalité pas tant que ça. Ces éléments normés sont ajoutés sous la forme de variables SCSS, ce qui fait qu'il est simple de les utiliser.

C'est même plus simple qu'avant : plutôt que de réfléchir à la couleur qu'on veut, on peut juste directement utiliser les variables, éventuellement en consultant la palette liée ci-dessus (qui permet d'ailleurs de copier les variables en un clic sur une couleur). Donc au lieu de deviner la couleur ou d'aller la rechercher et de jouer avec des `darken` pour l'ajuster, on fera plutôt : 

```scss
background-color: $primary-400;
```

Et si la couleur ne convient pas, pas de tâtonnements : on essaie `$primary-300` et `$primary-500` pour voir.

Pareil pour les ombres où c'est même encore plus simple : plus besoin de se souvenir de la syntaxe alambiquée des `box-shadow` quand ceci suffit.

```scss
@include shadow-2;
```

Si mis en place, l'objectif est qu'il soit **obligatoire** d'utiliser les couleurs provenant de la palette, sauf exceptions justifiées, par exemple nécessité de nouvelles teintes pour des fonctionnalités (provoquant l'extension de la palette afin de rester cohérent), ou encore pour des couleurs de marques (penser aux boutons de connexion via Facebook et Google). Et de même pour les tailles, ombres, marges, …

## Accessibilité du site

Notre site est loin d'être le pire en terme d'accessibilité, mais il a tout de même des progrès à faire. Notamment au niveau des contrastes qui sont souvent trop faibles. La réflexion sur les couleurs inclue donc une réflexion sur les contrastes (et autres sujets assimilés, par exemple le choix de couleurs distinguables par des daltoniens quand elles ont besoin de l'être). Par exemple, les cartes retravaillées des contenus ont un contraste nettement plus élevé afin d'être plus lisibles.

![Carte d'un contenu – avant](https://user-images.githubusercontent.com/1417570/85222790-3f935600-b3be-11ea-84a7-a02303673846.png)

![Carte d'un contenu – après](https://user-images.githubusercontent.com/1417570/91774194-8e444400-ebe8-11ea-9938-b11f2da9a0f0.png)

_Closes #2003 ; contribue à #4524._

### Contrôle qualité

Cette PR change a peu près toutes les couleurs et, de façon plus subtile, une bonne partie des longueurs, de ZdS. Il faut donc une QA de revue globale, de cohérence d'ensemble, et contre les détails oubliés dans cette grosse refactorisation.

1. Lancer le site en reconstruisant le _front_ (`make run`), puis aller sur une page au hasard et la recharger de force, afin d'avoir les dernières versions du CSS.
2. Visiter le site et vérifier que tous les éléments sont correctement affichés et lisibles. Plus subjectivement, s'assurer que les (relativement légers) changements visuels sont satisfaisants.

AmauryCarrade committed:
Import des normes depuis #5835

AmauryCarrade committed:
Ajustement de la palette

Ajout d'une couleur supplémentaire par catégorie et augmentation de la
luminosité des bleus des premières teintes primaires.

AmauryCarrade committed:
Première vague de normalisation du CSS

AmauryCarrade commented:
À noter que par rapport à #5835, la norme que je propose a un peu évolué : il y a plus de couleurs par catégorie (une de plus), les gris sont plus chaleureux (ce qui colle plus à l'image du site, je trouve), et les teintes primaires sont plus lumineuses dans les premières teintes (pour mieux refléter l'usage implicite qu'était fait avant la norme).

![Les primaires, avant/après](https://user-images.githubusercontent.com/1417570/91776138-0b71b800-ebed-11ea-8e03-49d49369369e.png)

![Les gris, avant/après](https://user-images.githubusercontent.com/1417570/91776180-23e1d280-ebed-11ea-8b3f-309823ab71b6.png)

AmauryCarrade committed:
Seconde vague de normalisation (normalement c'est terminé)

Avec quelques retouchessur la première vague au passage, et une légère
altération de la palette au niveau des oranges (couleurs d'accentuation).

Il ne manque que la documentation à récupérer de l'autre PR et à compléter.

AmauryCarrade commented:
Refactorisation terminée. Il manque encore la partie documentation que je ferai tantôt — pour cette raison, je laisse la PR en brouillon —, mais c'est déjà testable s'il y a des motivés. Les instructions de QA ont donc été mises à jour.

AmauryCarrade committed:
Bug dans le SCSS quand construit via make build-front

AmauryCarrade committed:
Retouches diverses

- Reprise de l'affichage des membres, en individuel et en liste
- Reprise de la zone métadonnée des contenus
- Autres détails ça et là

AmauryCarrade committed:
Reprise et finition de la documentation de #5835

La documentation commencée dans la PR #5835 a été récupérée et augmentée
avec toutes les normes déclarées et quelques conseils afin que le code
SCSS du site reste propre.

AmauryCarrade commented:
Voilà, cette fois c'est prêt !

- J'ai appliqué quelques menues retouches au site ça et là.
- J'ai repris et terminé la documentation de ces normes, et ajouté quelques conseils aux développeurs de l'interface pour que le code source reste propre et maintenable.

AmauryCarrade committed:
Retouche des listes d'utilisateurs.

Au passage, retouche de la page des membres par IP pour coller au CSS.

philippemilink commented:
Rapport de QA
--------------------

Tout d'abord, ce genre de PR n'est pas très évident à tester: il faudrait être capable de regarder toutes les pages et de *tout* regarder...

Je n'ai pas regardé le code, mais vu que c'est de la factorisation, ça me semble aller dans le bon sens, et il y a aussi de la doc, et ça c'est bien :wink: 

Mes remarques:
- sur la page d'accueil, le titre des articles / billets / tutoriels / sujets de forum n'est plus souligné quand on passe dessus. C'est voulu ?
- les champs de texte (par exemple pour modifier son pseudo et courriel) me semblent plus hauts, c'est normal ?
- (subjectif) autant je trouve que le contraste a bien été renforcé sur tout ce qui est listes de contenus, autant j'ai l'impression qu'il a baissé dans les formulaire, où je trouve le design un peu "fade"...

Je me suis baladé sur le site, je ne vois rien de choquant, ni de cassé. Je pense qu'il faudrait déployer cette PR sur la bêta pour faciliter les retours des utilisateurs (plutôt que d'avoir besoin d'une instance locale de ZdS pour tester...).

Sinon, quand je génère la doc, Sphinx n'est pas content (ce n'est pas bloquant):
```
/media/BigData/linux/Développement/zds/zds-site/doc/source/front-end/charte-interface.rst:68: WARNING: Analyse lexicale de literal_bloc impossible en "scss". Mise en évidence annulée.
/media/BigData/linux/Développement/zds/zds-site/doc/source/front-end/template-tags.rst:540: WARNING: Analyse lexicale de literal_bloc impossible en "guess". Mise en évidence annulée.
/media/BigData/linux/Développement/zds/zds-site/doc/source/front-end/template-tags.rst:557: WARNING: Analyse lexicale de literal_bloc impossible en "guess". Mise en évidence annulée.
/media/BigData/linux/Développement/zds/zds-site/doc/source/front-end/template-tags.rst:575: WARNING: Analyse lexicale de literal_bloc impossible en "guess". Mise en évidence annulée.
/media/BigData/linux/Développement/zds/zds-site/doc/source/front-end/template-tags.rst:594: WARNING: Analyse lexicale de literal_bloc impossible en "guess". Mise en évidence annulée.
/media/BigData/linux/Développement/zds/zds-site/doc/source/front-end/template-tags.rst:607: WARNING: Analyse lexicale de literal_bloc impossible en "guess". Mise en évidence annulée.
```
Je ne sais pas si c'est ce que tu évoquais hier sur Discord.

Je valide la PR, mais je ne merge pas.

AmauryCarrade commented:
Merci pour cette revue !

> Tout d'abord, ce genre de PR n'est pas très évident à tester: il faudrait être capable de regarder toutes les pages et de tout regarder...

Oui j'en suis conscient, ce pourquoi pour ce qui suivera j'essaierai de faire composant par composant.

> sur la page d'accueil, le titre des articles / billets / tutoriels / sujets de forum n'est plus souligné quand on passe dessus. C'est voulu ?

Hum, pas totalement, les tuiles de la page d'accueil sont en fait reprises de mon autre PR où il y avait un autre effet au survol rendant le soulignement superflu. Mais cet effet n'est pas présent ici. Faudrait-il le remettre (le soulignement), tu penses ?

> les champs de texte (par exemple pour modifier son pseudo et courriel) me semblent plus hauts, c'est normal ?

Oui, c'est volontaire. Ils sont plus grands donc plus facilement cliquables et plus agréable (les choses peuvent prendre de la place, c'est pas comme si on en manquait). Ça rentre dans la partie “accessibilité”.

> (subjectif) autant je trouve que le contraste a bien été renforcé sur tout ce qui est listes de contenus, autant j'ai l'impression qu'il a baissé dans les formulaire, où je trouve le design un peu "fade"...

Tu peux préciser ? Ça m'intéresse. Normalement, je n'ai pas trop touché aux formulaires, si ce n'est au niveau des tailles, donc, et des effets de _focus_ (qui manquaient). Les couleurs sont restées relativement les mêmes (au plus proche). Tout est encore modifiable, bien sûr !

> Je pense qu'il faudrait déployer cette PR sur la bêta pour faciliter les retours des utilisateurs (plutôt que d'avoir besoin d'une instance locale de ZdS pour tester...).

Je suis d'accord. @Situphen ? Peut-être après la bêta de la v29.3, ou avant vu qu'actuellement elle n'est pas prête ?

> Sinon, quand je génère la doc, Sphinx n'est pas content (ce n'est pas bloquant)

C'est pas exactement ce que je mentionnais, mais c'est lié. À noter qu'à l'exception du premier, ce sont des erreurs qui existent actuellement dans la documentation, sans cette PR. Cette PR en corrige d'ailleurs une ou deux au passage.

> Je valide la PR, mais je ne merge pas.

De toute façon, je ne crois pas qu'il était prévu de la prendre pour la v29.3, sauf si on change d'avis ^^ .

AmauryCarrade committed:
Corrige l'affichage des profils vides.

Le bloc bio doit prendre toute la largeur ; or ce n'était pas le cas.

Arnaud-D commented:
Salut. J'ai regardé rapidement, et le point qui me gêne vraiment est que le texte pour les cartes des tutos et articles est vraiment tassé sur la page d'accueil. Normalement l’œil devrait être attiré par le titre, mais là, ça fait juste gros paquet de texte.

Je ne sais pas ce qui serait le mieux pour arranger ça. Soit on peut jouer un peu sur les couleurs, soit sur les tailles relatives (les dates et auteur sont clairement des informations secondaires), soit en créant un peu plus d'espace avec des lignes plus hautes. Voilà, je me demande.

Ah et ça aussi : 
![image](https://user-images.githubusercontent.com/35631001/92995805-0bd05400-f507-11ea-9c8f-e936e9545b37.png)

AmauryCarrade commented:
Merci pour tes remarques ! Je vais re-regarder ça et essayer des choses pour améliorer la hiérarchie du composant. Jouer sur les couleurs c'est difficile : pour rester lisible, les textes oranges ne peuvent guère être éclaircis… sans passer sous le seuil de lisibilité. Sauf à changer complètement les couleurs pour une plus contrastée (j'avais tenté le bleu de ZdS) mais quand j'avais proposé ça avec l'autre PR, ça n'avait été que peu apprécié.

Peut-être aussi jouer sur la graisse. C'est plus élégant que sur les tailles, je trouve.

Concernant le second point — bien vu aussi, ça devrait être facile à corriger :) .

firm1 commented:
Bonjour,

J'ai effectué mon test sur la version disponible en beta le 10/10/2020.

Pour info, je ne suis pas malvoyant.

J'ai essayé de structurer un peu les catégories de mes retours, je n'ai pas tout testé, mais voilà les premiers.

## Les manques de cohérence

### boîtes customs : la couleur du bloc vs la couleur de l'icone

![Capture d’écran de 2020-10-10 14-51-36](https://user-images.githubusercontent.com/6066015/95655802-25ff5100-0b0a-11eb-8cf1-bae4e2d4c251.png)


On remarque ici que la couleur de l'icone "i" qui symbolise l'information a une couleur différente de la couleur tu titre du texte.

Il faudrait idéalement modifier l'icone d'info pour retrouver la cohérence.

On a le même problème avec le bloc attention.

### Le changement du orange n'est pas propagé partout

Le orange a peut être été changé sur la page d'accueil, mais on retrouve encore l'ancien orange sur les zones ci-dessous (ça ne fait pas forcément harmonieux)

![Capture d’écran de 2020-10-10 15-04-23](https://user-images.githubusercontent.com/6066015/95655783-f51f1c00-0b09-11eb-801d-54f0e84b61eb.png)
![Capture d’écran de 2020-10-10 15-04-05](https://user-images.githubusercontent.com/6066015/95655786-f7817600-0b09-11eb-9fbe-7afc66bd1568.png)
![Capture d’écran de 2020-10-10 15-03-26](https://user-images.githubusercontent.com/6066015/95655788-fa7c6680-0b09-11eb-9c78-40f298d7156d.png)

Il faudrait garder un seul orange sur la page d'accueil et le décliner uniquement pour exprimer une différence de ton.

## Les changements que je ne trouve pas agréable

### Le orange qui vire au rouge

![Capture d’écran de 2020-10-10 14-56-06](https://user-images.githubusercontent.com/6066015/95655808-30214f80-0b0a-11eb-830c-764ea940c260.png)


Je ne suis clairement pas fan du fait que orange d'avant vire au rouge maintenant. ça fait un peut terne je trouve.

J'ai fait quelques essais, et j'arrive a obtenir un truc que je trouve moins terne avec cet orange `#ff6d00`

![Capture d’écran de 2020-10-10 15-01-59](https://user-images.githubusercontent.com/6066015/95655811-36173080-0b0a-11eb-8ac6-353b311cbec7.png)

AmauryCarrade commented:
Merci @firm1 ! Je vais essayer d'adresser tout ça. Pour les couleurs, la difficulté c'est que les oranges trop lumineux (qui tendent vers le jaune) sont plus difficiles à rendre accessibles. D'où le choix de les faire tendre vers le brun dans les teintes sombres. Mais je vais essayer d'ajuster pour voir ce que ça peut donner :) .

Sur les oranges qui ne sont pas identiques partout c'est une erreur, ça ne devrait pas être le cas (accent 500 partout normalement, sur ces lignes !)

Pour les icônes, la difficulté est que les icônes actuelles sont des images PNG (!) d'assez basse qualité et à couleur fixe, donc difficile de les faire évoluer. Mais j'ai pour intention de les reprendre directement en CSS (ou à défaut en SVG).

firm1 commented:
Petit bug sur mobile.

![Screenshot_20201124-124957_Brave](https://user-images.githubusercontent.com/6066015/100090685-d83d6e80-2e53-11eb-9b00-077f8d949157.jpg)


Avec les tags a droite,  la partie de gauche fait assez tassée et on a une impression de perte d'espace.

AmauryCarrade commented:
Merci !

AmauryCarrade committed:
Amélioration des fiches de contenus

- Fiches moins denses sur l'accueil avec le retrait des catégories
  (dont l'intérêt laisse à désirer sur l'accueil).
- Reprise du CSS afin de mieux gérer le positionnement des différents
  éléments (en particulier les métadonnées).
- Correction d'un bug d'affichage où les appareils avec une haute
  densité de pixels voyaient une image de fond sans aucun rapport.

AmauryCarrade committed:
Retour du soulignement au survol des fiches de contenus.

AmauryCarrade committed:
Amélioration (tentative) de la zone de nouveau tuto sur l'accueil

- Le texte n'est plus “délavé”.
- Nouvel effet au survol, plus marquant qu'un changement de
  0,2% de teinte.

AmauryCarrade committed:
Pied de page manquant de marges

AmauryCarrade committed:
Correction du titre des modales par dessus les icônes

AmauryCarrade committed:
Correction de l'alignement des icônes des titres

AmauryCarrade committed:
Retrait d'un zindex inutile

Qui en plus faisait passer le texte au dessus de certains menus

AmauryCarrade committed:
Correction de marges

AmauryCarrade committed:
Mauvais affichage de l'en-tête des contenus sur mobile

AmauryCarrade commented:
J'ai avancé là dessus en traitant ce qu'on m'a dit ici, sauf : 

- l'orange/jaune, il faut que j'y réfléchisse ; et
- les icônes, idem, j'hésite entre mettre ces icônes à jour ou les refaire en CSS ou SVG ce qui règlerait le souci de couleurs bien plus radicalement.

Aussi le souci de couleurs sur les lignes n'en est… pas un, elles sont toutes bonnes. Par contre j'utilise (volontairement) un jaune plus foncé pour les textes (sinon c'est illisible), mais ça reste la même teinte.

Situphen commented:
J'ai mis à jour la bêta avec les derniers changements !

AmauryCarrade commented:
Ce qui me permet de noter ce bug quand il y a trop d'auteurs

![image](https://user-images.githubusercontent.com/1417570/100477667-464e9380-30e9-11eb-920f-b715e6077f46.png)

Situphen commented:
Je suis allé faire un tour sur la bêta pour regarder un peu toutes les pages et mis à part les quelques points que je décris ci-dessous le rendu est très bien !

Barre de navigation : 

- La couleur de fond au survol des boutons (« Bibliothèque », « Forum », MPs, notifications...) n'est pas assez constatée par rapport à la couleur de fond par défaut à mon avis (difficile de voir si on survol ou pas ces boutons là)
- Idem pour les liens du menu déroulant de l'avatar (`.my-account-dropdown .dropdown-list`)
- À l'inverse, il y a un trop gros contraste pour la couleur de fond au survol des catégories de contenus (`.dropdown-list-element`) à mon avis (ça fait mal aux yeux je trouve)

Sur la page d'accueil : 

- La pointe de la bulle des commentaires (`.content-reactions::before`) n'a pas la même couleur de fond ni la même ombre que le rond de la bulle des commentaires (`.content-reactions`)
- Le bloc pour inciter à publier un contenu chez nous (`.write-tutorial`) à une bordure grise au lieu d'avoir une bordure bleu (même bleu que la couleur de fond)
- Le bouton « Commencer à rédiger » (`.btn-write-tutorial`) n'a pas exactement les même couleurs de texte et de fond par défaut que lors du survol de la souris (`:hover`) : 
  - soit on met les mêmes couleurs (et l'effet au survol est mis en évidence simplement par le changement de taille) ;
  - soit on choisit des couleurs bien distinctes.
- Les blocs de contenus ont du orange alors que les blocs du forum ont du bleu, il faudrait à mon avis uniformiser ça (soit orange, soit bleu)
- Dans les blocs du forum (`.topic-item`), les liens vers les membres (`a.member-item`) sont survolés en gris lors du survol de la souris (`:hover`)
- À mon avis il faudrait enlever l'arrondi sur les tags (`.content-item .content-tags`) et le bouton « Commencer à rédiger » (`.btn-write-tutorial`) car tout les autres boutons n'ont pas d'arrondis

Tribune : 

- La bulle des commentaires est martyrisée par le texte en orange lorsque celui-ci est trop long
  ![Capture d'écran du soucis](https://user-images.githubusercontent.com/6664636/101828767-4495be80-3b32-11eb-8f26-b8a091dac31c.png)

Divers : 

- Le fond bleu transparent lorsqu'on ouvre une modale (`.modals-overlay`) est un peu trop bleu (ça fait bleu+bleu donc c'est écœurant) à mon goût (peut-être que garder du noir serait plus raisonnable)

Messagerie : 

- 
![Capture d'écran](https://user-images.githubusercontent.com/6664636/102028063-83cd4500-3da8-11eb-83f0-c9ec861bfbde.png)

AmauryCarrade committed:
Modification des noirs et de la couleur des contenus

- Le noir impur est à peine plus clair.
- Les contenus sont affichés en noir et non en gris sombre, améliorant
  ainsi le confort de lecture.

AmauryCarrade committed:
Ajustement des couleurs de l'en-tête

- La couleur du survol des liens et du contenu des blocs
  déroulés est plus claire.
- La couleur au survol des liens dans les blocs déroulés
  a été ajustée afin que le contraste soit moins violent.

AmauryCarrade committed:
Correction de la bordure de l'incitation de rédaction.

- Elle était grise. Elle est bleue. Parfois, bordure
  évolue.

AmauryCarrade committed:
Tous les pieds de contenus sur l'accueil sont orange.

- Même les sujets.
- Fin de la différenciation.
- Tout le monde à la même enseigne.

AmauryCarrade committed:
Contenus sur l'accueil : seul le titre est souligné

- Le sous-titre ne l'est plus.
- C'est moins trompeur.
- Le côté cliquable est toujours mis en avant par le
  soulignement du titre.
- Moins de soulignement c'est toujours une bonne nouvelle.

AmauryCarrade committed:
Attention ça pique.

- Les arrondis sont pointus.
- Cohérence avec le reste.
- Ça reviendra peut-être.
- Attention aux doigts.

AmauryCarrade committed:
La bulle de commentaires est sauvée.

- Fin du harcèlement de la bulle par la liste des
  auteur⋅e⋅s.
- Elle était écrasée, elle ne l'est plus.
- S'il y a trop d'auteur⋅e⋅s (je vous vois, les récaps),
  iels s'abbrègeront.

AmauryCarrade committed:
Un fond moins sombre pour les modales.

- Le fond des modales fait polémique.
- On tente des choses.
- Expérimental, pourra toujours évoluer.

AmauryCarrade committed:
Correction du bouton d'ajout d'un membre à un MP.

- Lui aussi tentait l'art moderne.
- Là non plus, pas un succès.
- Il est plus sage.

AmauryCarrade commented:
> * La pointe de la bulle des commentaires (`.content-reactions::before`) n'a pas la même couleur de fond ni la même ombre que le rond de la bulle des commentaires (`.content-reactions`)

Étrange : le CSS est formel, c'est la même couleur.

> * Le bouton « Commencer à rédiger » (`.btn-write-tutorial`) n'a pas exactement les même couleurs de texte et de fond par défaut que lors du survol de la souris (`:hover`) :       
>   * soit on met les mêmes couleurs (et l'effet au survol est mis en évidence simplement par le changement de taille) ;
>   * soit on choisit des couleurs bien distinctes.

J'ai choisi la première solution.
 
> * Les blocs de contenus ont du orange alors que les blocs du forum ont du bleu, il faudrait à mon avis uniformiser ça (soit orange, soit bleu)

Passons tout en orange.

> * Dans les blocs du forum (`.topic-item`), les liens vers les membres (`a.member-item`) sont survolés en gris lors du survol de la souris (`:hover`)

C'est du passé. Ils ont voulu s'essayer à l'art moderne mais c'était pas un succès. (En vrai ça venait d'un débordement du fait qu'on utilise le même style pour les blocs contenant avatar + pseudo, qui eux ont un fond au survol.)
 
> * À mon avis il faudrait enlever l'arrondi sur les tags (`.content-item .content-tags`) et le bouton « Commencer à rédiger » (`.btn-write-tutorial`) car tout les autres boutons n'ont pas d'arrondis

Ça reviendra :eyes: mais pas tout de suite, effectivement.

> * La bulle des commentaires est martyrisée par le texte en orange lorsque celui-ci est trop long
>   ![Capture d'écran du soucis](https://user-images.githubusercontent.com/6664636/101828767-4495be80-3b32-11eb-8f26-b8a091dac31c.png)

Zeste de Savoir s'oppose à toute discrimination ou forme de harcèlement. Ce comportement a donc été corrigé.

>  * Le fond bleu transparent lorsqu'on ouvre une modale (`.modals-overlay`) est un peu trop bleu (ça fait bleu+bleu donc c'est écœurant) à mon goût (peut-être que garder du noir serait plus raisonnable)

J'ai tendance à préférer comme ça, trouvant que ça donne un peu plus d'identité… Je vais tenter une réduction, et faire un sondage pour voir ce qui est plébiscité.

Le reste est traité. J'ai aussi changé la couleur des textes afin qu'ils soient plus agréables à lire (cf. commits). N'hésitez pas à faire des retours sur la bêta !

Situphen commented:
> Étrange : le CSS est formel, c'est la même couleur.

En fait c'est le `filter: drop-shadow(0 -1px 3px rgba(15,15,15,.1));` qui donne une ombre sur la flèche, ce qui crée une séparation entre le rond et la flèche.

**Capture d'écran avec cette propriété**

![Capture d'écran avec cette propriété](https://user-images.githubusercontent.com/6664636/102691697-a0390980-420e-11eb-9a59-332625608c1f.png)

**Capture d'écran sans cette propriété**

![Capture d'écran sans cette propriété](https://user-images.githubusercontent.com/6664636/102691737-eee6a380-420e-11eb-9347-39c2c74ac708.png)

Est-ce qu'on ne pourrait pas plutôt différencier la bulle et la flèche du reste de l'encart avec une couleur de fond différente plutôt qu'une ombre ?

AmauryCarrade commented:
On peut, ou encore faire les deux. Mais quelle couleur ?

Situphen commented:
Je viens de trouver la solution ! Il faut supprimer le `box-shadow` de `.content-reactions` et le `filter` de `.content-reactions::before` et mettre un `filter` directement sur `.content-reactions`. Je te laisse trouver les bons paramètres pour celui-ci mais en tous cas ça permet d'avoir une belle bulle !

![Capture d'écran](https://user-images.githubusercontent.com/6664636/102693083-40476080-4218-11eb-8d4d-1469c117e0b9.png)

AmauryCarrade commented:
Ah bien vu ! Je tente ça :)

AmauryCarrade committed:
Meilleur affichage des bulles.

- Les bulles se montrent dans toute leur splendeur.
- Leur pointe n'est plus injustement censurée.
- Même le fond blanc ne leur fait pas peur et ne les
  masque plus. #BlackLivesMatter
- L'ombre est la première standard, mais assombrie pour
  compenser le fait qu'on ne peut en mettre deux.

AmauryCarrade commented:
Voilà ! Ça donne ça, je trouve que c'est propre. N'hésitez pas si vous n'êtes pas d'accord.

![La bulle dans toute sa splendeur ombrée](https://user-images.githubusercontent.com/1417570/102694111-4725a180-421f-11eb-8a34-a9d1ac29bd2c.png)

AmauryCarrade commented:
A priori, il ne me reste qu'à ajuster les icônes des blocs de couleur. Il me faudrait une version SVG de ces icônes (sans fond), pour faire quelque chose de propre au passage.

Ainsi qu'à ajuster la version mobile (quelques détails) et des blocs de code en ligne qui sont améliorables.

Plus tout ce que vous pourrez dire ^^ .

philippemilink commented:
J'ai mis à jour la bêta, qui contient maintenant les derniers commits poussés.

artragis commented:
a priori le bug relatif à la longueur de la description des auteurs est encore présent 
![image](https://user-images.githubusercontent.com/1523653/102699391-9205e000-4244-11eb-8583-3ad2c531bb07.png)

Situphen commented:
> a priori le bug relatif à la longueur de la description des auteurs est encore présent
image

Quand je vais sur la bêta je n'ai plus le bug personnellement. Peut-être un soucis de cache ?

![Capture d'écran de la bêta](https://user-images.githubusercontent.com/6664636/102699635-82879680-4246-11eb-8a57-204f6247280d.png)

-----

Un truc qui me chiffonne c'est que je trouve les couleurs du menu déroulant beaucoup trop flashy par rapport à celle actuellement utilisées, ça m'agresse les yeux. :/ Est-ce que vous avez la même impression ?

Arnaud-D commented:
Je n'ai plus le problème de dépassement.

---

Les couleurs sont effectivement plus saturée/flashy à certains endroits, mais c'est peut être juste l'habitude de la sobriété actuelle qui fait que ça ressort.

---

L'écart à la bordure sur les cartes donne une impression d'irrégularité. Au sein d'une même carte, mais aussi entre cartes. C'est un peu dur à expliquer, c'est une espèce de sensation d'inhomogénéité. On a déjà un peu ce problème actuellement, mais c'est beaucoup moins flagrant.

![image](https://user-images.githubusercontent.com/35631001/102699811-26be0d00-4248-11eb-9226-09ef251ce204.png)

![image](https://user-images.githubusercontent.com/35631001/102699832-540abb00-4248-11eb-96eb-6402c0ad06ed.png)

![image](https://user-images.githubusercontent.com/35631001/102700124-ffb50a80-424a-11eb-86e7-4c90aea0a7f6.png)

---

Ensuite, je trouve que la hauteur de ligne (ou l'interligne) est faible sur les tutos, et que ça gêne un peu la lecture. J'ai 23.8px de hauteur de ligne sur le site actuel et sur cette PR on est à 21px pour une taille de police de 14px dans les deux cas. Le ratio de la PR est à 1.5 (contre 1.7 avant). Je ne pense pas que ce soit tellement le ratio, mais l'association du ratio et de la taille. J'ai testé des sites connus où je trouve que ça va : 16px@1.5, 18px@1.55,  14px@1.6. Le mode lecture de Firefox me propose du 20px@1.6

J'aurai tendance à dire que 1.6 de ratio c'est pas mal quand je suis à 14px sur la police. Le ratio pouvant être plus faible à mesure que la police est grande. En tout cas c'est mon ressenti.

AmauryCarrade commented:
Tout à fait, l'interligne est clairement trop faible par rapport à la taille du texte (c'est toujours proportionnel aux deux ça de toute façon). Bonne remarque.

Idem pour tout le reste, je regarderai ça. Pour les couleurs, j'adapterai peut-être un peu la palette pour avoir des couleurs un peu plus sobres, sans non plus virer dans l'excès inverse. Enfin, je tenterai.

artragis commented:
> Je n'ai plus le problème de dépassement.

Je l'ai sur la béta en non connecté

AmauryCarrade commented:
> > Je n'ai plus le problème de dépassement.
> 
> Je l'ai sur la béta en non connecté

Que ce soit connecté ou déconnecté, en version normale ou petite, sur émulateur ou sur vrai téléphone (Firefox Mobile Nightly (71)), je n'arrive pas à reproduire. Tu es sous quelle configuration ?

artragis commented:
Ok, sur mon linux, avec le dernier firefox j'avais le soucis, sur mon windows, ça passe crème.

AmauryCarrade committed:
Léger ajustement des couleurs primaires

- La couleur primaire 400 est un peu moins “flashy” (bon, à peine,
  hein).

AmauryCarrade committed:
Amélioration du CSS des boîtes de contenu.

- Variabilisation des parties communes et nettoyage.
- Les billets ont des dates.

AmauryCarrade committed:
Meilleure lisibilité des contenus.

- La hauteur des lignes a été augmentée à 1.6.

AmauryCarrade committed:
Ajustement des bulles d'informations aux nouvelles couleurs

AmauryCarrade committed:
Ajustement des coches et des pouces aux nouvelles couleurs

AmauryCarrade committed:
Retour des marges entre blocs sur les grands écrans

- Une règle sur les grands écrans avait des effets de bord en
  supprimant beaucoup trop de marges de façon très violente
  (ce !important du démon…).

AmauryCarrade committed:
Titres plus compacts

- Les titres des contenus étaient démesurément espacés, dés qu'ils
  passaient sur plusieurs lignes.

AmauryCarrade commented:
J'ai un peu repris tout ça. Les commits parlent d'eux même pour ce qui a été fait.

> L'écart à la bordure sur les cartes donne une impression d'irrégularité. Au sein d'une même carte, mais aussi entre cartes. C'est un peu dur à expliquer, c'est une espèce de sensation d'inhomogénéité. On a déjà un peu ce problème actuellement, mais c'est beaucoup moins flagrant.

@Arnaud-D J'ai regardé ça, et en réalité, on est déjà sur quelque chose d'assez cohérent : toutes les mesures sont les mêmes partout (j'ai d'ailleurs modifié le CSS pour que ça reste le cas). La seule exception est en cas de présence d'une bulle de commentaire, mais si l'alignement est maintenu, la bulle est décalée, ce qui fait bizarre. Aurais-tu une meilleure solution ?

> Les couleurs sont effectivement plus saturée/flashy à certains endroits, mais c'est peut être juste l'habitude de la sobriété actuelle qui fait que ça ressort.

@Arnaud-D @Situphen J'ai à peine altéré les couleurs (la 400) pour que ce soit moins flashy. Mais je n'arrive pas à savoir si c'est vraiment un _problème_. À titre personnel, ça ne me dérange pas.

![Évolution de la palette (teinte primaire 400)](https://user-images.githubusercontent.com/1417570/103141207-8f4a3580-46f1-11eb-8f2b-746070598265.png)

> J'aurai tendance à dire que 1.6 de ratio c'est pas mal quand je suis à 14px sur la police. Le ratio pouvant être plus faible à mesure que la police est grande. En tout cas c'est mon ressenti.

Je suis passé à 1.6, c'est effectivement bien mieux. À terme (si je continue dans ce que je proposais pour une nouvelle interface pour ZdS), je me demande si on ne pourrai pas aussi grossir le texte lui-même (comme ce que fait le mode lecture de Firefox, même si peut-être pas _à ce point_). J'ai également réduit l'interligne des titres, qui elle, était énorme.

Situphen commented:
@philippemilink Est-ce que tu peux mettre à jour la bêta pour que l'on puisse voir les changements ?

> J'ai à peine altéré les couleurs (la 400) pour que ce soit moins flashy.

Difficile de voir avec les petits carrés si ça va rendre moins flashy sur le site ou pas.

> on est déjà sur quelque chose d'assez cohérent : toutes les mesures sont les mêmes partout (j'ai d'ailleurs modifié le CSS pour que ça reste le cas)

Sur ce point, il faut faire attention car ce n'est pas nécessairement parce que les mesures sont les mêmes partout que cela donne une impression de cohérence (qui est ce que l'on recherche ici). Un bon exemple est à mon avis la Tour Eiffel : pour donner l'impression qu'il y a la même couleur du bas jusqu'en haut, ils sont obligés d'utiliser différentes teintes de couleur.

AmauryCarrade commented:
> Sur ce point, il faut faire attention car ce n'est pas nécessairement parce que les mesures sont les mêmes partout que cela donne une impression de cohérence (qui est ce que l'on recherche ici).

@Situphen Certes oui, c'est d'ailleurs déjà pris en compte : la marge en bas est légèrement plus grande (par une marge ajoutée au `footer`, de mémoire) que celle d'en haut pour compenser la différence de taille.

philippemilink commented:
> @philippemilink Est-ce que tu peux mettre à jour la bêta pour que l'on puisse voir les changements ?

Done ! Rafraîchissez vos caches ! 😃

Situphen commented:
Le changement de couleur est en effet très faible mais j'ai bien l'impression que c'est moins flashy dans cette nouvelle version !

Ton dernier commit, qui sauf erreur de ma part visait les titres au sein des contenus, a aussi modifié les titres du sommaire et je pense que pour ceux là il faudrait mettre un petit coup de `padding` pour revenir comme avant ! :)

![Capture d'écran](https://user-images.githubusercontent.com/6664636/103142968-823b3f80-470d-11eb-8a46-18d65f912f1e.png)

Les encarts de la recherche sont cassés [visiblement](https://beta.zestedesavoir.com/rechercher/?q=foo&category=&subcategory=&from_library=).

À part ça je n'ai rien remarqué d'anormal (mais je continuerai de creuser car il doit rester des coquilles).

AmauryCarrade commented:
> Le changement de couleur est en effet très faible mais j'ai bien l'impression que c'est moins flashy dans cette nouvelle version !

J'ai littéralement diminué la luminosité de la teinte d'un unique pourcent :D 

> Ton dernier commit, qui sauf erreur de ma part visait les titres au sein des contenus, a aussi modifié les titres du sommaire et je pense que pour ceux là il faudrait mettre un petit coup de `padding` pour revenir comme avant ! :)

Effectivement ! Et ça s'applique aussi aux titres de niveau un au sein des contenus, en réalité.

> Les encarts de la recherche sont cassés [visiblement](https://beta.zestedesavoir.com/rechercher/?q=foo&category=&subcategory=&from_library=).

Bien vu, un effet de bord (>_>) auquel je n'avais pas pensé

> À part ça je n'ai rien remarqué d'anormal (mais je continuerai de creuser car il doit rester des coquilles).

Merci beaucoup pour tes recherches de problèmes ! Ça m'aide énormément. De mon côté, j'ai noté quelques points que je voulais encore travailler, notamment les marges bizarres dans les sommaires des contenus (notamment les gros tutos), et les bandeaux de signalement qui ne sont pas très propres (je pense les reprendre complètement, ça ira très vite vue la complexité, et ça cassera une interdépendance de styles indésirable).

AmauryCarrade committed:
Meilleures marges dans les grands titres de contenus

AmauryCarrade committed:
Correction du sommaire des tutoriels

AmauryCarrade committed:
Reprise des alertes

- Plus lisibles
- Plus jolies
- Découplées des bandeaux de notification !

AmauryCarrade committed:
Correction de l'affichage de la recherche.

- Les résultats de la recherche n'avaient pas été convertis au
  nouveau format des cartes.

AmauryCarrade commented:
Voilà ! J'ai repris ce qu'a été signalé ou que j'avais noté (marges, recherche, sommaires, alertes). Concernant les alertes, j'ai, comme je l'annonçais, découplé ces dernières des bandeaux du site, ce qui a permis au passage de plus facilement adapter leur mise en page à l'usage réel qu'on en fait, notamment en rendant mieux les longs messages d'alerte.

![Aperçu des alertes](https://user-images.githubusercontent.com/1417570/103156303-aa777c80-47a7-11eb-9b25-cd464d1b7798.png)

Ça devient… vraiment prêt ?

@philippemilink (déso) si t'as le temps de le mettre en bêta ce serait génial, pour les dernières vérifications approfondies !

philippemilink commented:
> @philippemilink (déso) si t'as le temps de le mettre en bêta ce serait génial, pour les dernières vérifications approfondies !

Fait. Pas de quoi être désolé, c'est vraiment rapide à faire (❤️ sur ceux qui ont conçu tout le système Ansible !), donc si je suis devant mon ordi c'est vraiment pas un problème 😃 

J'en profite pour faire quelques retours:
- les boîtes _information_ sont un peu trop... "bleues": j'avais l'impression que j'avais sélectionné le texte ! (par exemple: https://beta.zestedesavoir.com/tutoriels/885/introduction-a-la-preuve-de-programmes-c-avec-frama-c-et-son-greffon-wp/)
- en bas de https://beta.zestedesavoir.com/billets/ la barre de pagination est un peu trop collée aux billets...
- https://beta.zestedesavoir.com/membres/voir/philippemilink/: la ligne "dernière réponse" est tronquée pour le sujet "CreateQueryBuilder: utiliser la fonction UPPER()" que j'avais créé
- sinon je me suis baladé un peu partout sur le site, comme un utilisateur "lambda" et je n'ai rien trouvé d'autres qui serait cassé ou gênant.

AmauryCarrade committed:
Correction des marges entre blocs sur l'accueil

AmauryCarrade committed:
Correction de dépassements dans les extraits de forum

- Dépassements sur la page d'accueil et les pages de profil sur les
  extraits de forum, sur les très petits écrans

AmauryCarrade committed:
Diminution de l'interligne de la citation de l'accueil

AmauryCarrade committed:
Toutes les alertes utilisent le même modèle graphique.

- Déplacement de la logique d'affichage des alertes dans un unique
  gabari.
- Ce gabari permet de spécifier l'URL à utiliser pour résoudre.
- Les contenus, les profils, et les messages, utilisent désormais
  ce gabari.

AmauryCarrade committed:
Correction des marges du bandeau de pagination.

- Lorsqu'un contenu ne se termine pas par un paragraphe (par exemple,
  par une vidéo), le bandeau de pagination n'est plus collé au
  contenu.

AmauryCarrade committed:
La page n'est plus trouée.

- Retrait de la marge au dessus du pied de page, qui n'était pas
  vraiment utile, et qui faisait apparaître, sur mobile, le menu
  latéral caché sous le conteneur principal (`.main-container`).
- Correction de marges incorrectes des alertes sur mobile.

AmauryCarrade committed:
Les notifications étaient mal alignées

- Suite à un effet de bord, les éléments de la liste des notifications
  (au sens large : alertes, MPs ou notifications) étaient mal alignées.
- Le bloc flottant des notifications était mal positionné sur tablette.

AmauryCarrade committed:
Correction de l'alignement de l'en-tête sur mobile

AmauryCarrade committed:
Les blocs d'information sont plus clairs.

En conséquence les mini-codes dans ces blocs sont plus sombres.

AmauryCarrade commented:
Normalement, toutes les remarques qui m'ont été faites par @philippemilink ci-dessus et par @Situphen en privé sont corrigées. Moyen de mettre la bêta à jour ? :)

AmauryCarrade committed:
Correction de l'alignement des titres des modales sur mobile

AmauryCarrade committed:
Correction de l'affichage de la liste des participants d'un MP

- Aussi, corrige l'affichage de l'auteur d'un sujet du forum

philippemilink commented:
Bêta à jour.

Mes remarques ont effectivement été corrigées. 👍

AmauryCarrade committed:
Correction de la position des menus de notifications

artragis commented:
Petit changement que je viens de détecter : quand ou regarde le détail des votes sur un poste et qu'il y a deux personnes ou plus, avant, au passage de la souris, la zone où se trouve le pseudo qu'on survole changeait de luminosité en fond, ce n'est plus le cas maintenant.

artragis commented:
Lorsqu'on signale un profile, l'espace avant la textbox a été agrandi 

*v27.1*
![Avant](https://user-images.githubusercontent.com/1523653/103177238-08c06000-4879-11eb-935b-474ab941f1d1.png)

*CSS normalisé*
![image](https://user-images.githubusercontent.com/1523653/103177276-576dfa00-4879-11eb-8af1-5ef4c0410530.png)

a priori il y a eu 15px de plus entre le label et le haut de la boîte

artragis commented:
Autre petit bug : lorsqu'un tuto a nécessité plus d'un contributeur, la liste des contributeurs affiche un tableau avec au moins une ligne "claire". Lorsqu'on survole le pseudo, celui-ci prend un couleur de fond étrange par rapport à ce qu'on observe lorsque la ligne est une ligne sombre.

AmauryCarrade committed:
Correction de l'affichage de la liste des contributeurs

L'objectif initial était de corriger un effet de survol incorrect dans
feu le tableau, mais j'ai pris la liberté d'utiliser le nouveau
composant de liste d'utilisateurs, ce qui permet un meilleur rendu
sans passer par un artificiel tableau.

AmauryCarrade committed:
Correction des marges dans les boîtes modales

- Correction des marges générales, qui n'étaient pas très cohérentes
- Correction des marges dans le cas spécifique de formulaires qui ont
  une marge séparatrice superflue dans certains cas.

AmauryCarrade commented:
> Petit changement que je viens de détecter : quand ou regarde le détail des votes sur un poste et qu'il y a deux personnes ou plus, avant, au passage de la souris, la zone où se trouve le pseudo qu'on survole changeait de luminosité en fond, ce n'est plus le cas maintenant.

@artragis Je constate le contraire (ci-dessous, ma souris est sur le second membre). Peut-être la différence est-elle insuffisante ?

![Survol sur le second élément du tableau des personnes ayant aimé un élément](https://user-images.githubusercontent.com/1417570/103177769-89815b00-487d-11eb-917e-084d9192141a.png)


> Autre petit bug : lorsqu'un tuto a nécessité plus d'un contributeur, la liste des contributeurs affiche un tableau avec au moins une ligne "claire". Lorsqu'on survole le pseudo, celui-ci prend un couleur de fond étrange par rapport à ce qu'on observe lorsque la ligne est une ligne sombre.

Réglé. J'en ai profité pour utiliser à la place du tableau un élément introduit dans cette PR (enfin, normalisé, plutôt).

![Liste des contributeurs, sous la forme d'une liste de membres et non d'un tableau](https://user-images.githubusercontent.com/1417570/103178169-c13dd200-4880-11eb-9126-89633f174fd2.png)

Le reste est réglé aussi.

philippemilink commented:
Bêta à jour.

AmauryCarrade committed:
Support pour les listes de membres en pleine largeur.

AmauryCarrade committed:
Ajout de la documentation pour les listes de membres.

AmauryCarrade committed:
`.authors` → `.members`

C'est beaucoup plus logique vu que c'est utilisé pour plein d'autres
choses.

AmauryCarrade commented:
> Réglé. J'en ai profité pour utiliser à la place du tableau un élément introduit dans cette PR (enfin, normalisé, plutôt).

Encore plus cohérent avec un nouveau mode “pleine largeur” pour cette version refactorisée des listes de membres — désormais documentée.

![Liste de membres avec qualification, en pleine largeur](https://user-images.githubusercontent.com/1417570/103181180-abd7a080-489e-11eb-9fd2-6f48340f74c7.png)

AmauryCarrade committed:
Documentation de l'affichage des alertes

Aussi, renommage du paramètre `solve_url` en `alert_solve_url`, par
cohérence.

AmauryCarrade committed:
Correction des liens vers la norme dans la documentation

artragis commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,250 @@
+========================================
+
+Les styles de Zeste de Savoir sont complexes, formés de l'empilement d'années
+de contributions par des personnes très différentes. Pour qu'il reste maintenable,
```
ils restent (puisqu'on parle des styles)

artragis commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,250 @@
+composent : couleurs, polices, tailles, ombres…
+
+Tout développement graphique sur Zeste de Savoir **doit** respecter cette norme
+(avec quelque exceptions mentionnés ci-après).
```
mentionnées

artragis commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,250 @@
+   que parmi une liste restreinte de possibilités permet de gagner du temps :
+   on ne passe plus deux heures à essayer toutes les valeurs possibles entre
+   deux tailles ou deux couleurs ; on prend la plus proche qui rend bien et
+   *voilà*. Aussi, ça évite de *deviner* quelle valeur choisir au hasard.
```
Troisième argument : les valeurs ont été étudiées pour faciliter l'accessibilité globale du site.

artragis commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,250 @@
+Polices
+-------
+
+Zeste de Savoir utilise trois polices de caractère sur le site :
```
caractères

artragis commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,250 @@
+
+Zeste de Savoir utilise trois polices de caractère sur le site :
+
+- une police sans empattements (``$font-sans-serif``), pour les textes de l'interface (il s'agit
```
Pas de s en français quand le quantificateur est 0

philippemilink commented:
Bêta à jour.

artragis commented on doc/source/front-end/interface-utilisateur.rst:
```diff
@@ -109,8 +109,8 @@ Voici un exemple.
 
-  <a href="#modal" class="open-modal edit-button" title="{% trans "Modifier la licence" %}">
-    <span class="visuallyhidden">{% trans "Modifier la licence" %}</span>
+  <a href="#modal" class="open-modal edit-button" title="Modifier la licence">
```
Pourquoi avoir retiré l'i18n?

AmauryCarrade replied:
C'est clairement une erreur, je corrigerai ça

artragis commented on templates/gallery/gallery/details.html:
```diff
@@ -118,26 +118,28 @@
-        <button class="toggle-gallery-view btn btn-grey" type="button" title="Alterner entre les modes de vue grille et liste">
-            {% trans "Vue grille" %}
-        </button>
+        <header class="topics-list-header">
```
sémantiquement header est normal? c'est pas plutôt aside ou nav dans ce genre de situation?

AmauryCarrade replied:
Sémantiquement, `<header>` représente une en-tête même d'un sous composant. Par exemple, utiliser `<header>` pour l'entête des messages ce serait ok. Donc a priori c'est bon.

Je vérifierai plus en profondeur quand j'aurai accès à tout le code (le site mobile ne me montre que l'extrait, le vilain).

AmauryCarrade replied:
Ok, avec le reste du code, `<aside>` serait tout de même plus logique.

AmauryCarrade replied:
Ok, avec le reste du code, `<aside>` serait tout de même plus logique.

artragis reviewed (changes requested):
Ce travail est ouf. Je pense qu'il est suffisament mature pour faire face à des chipotages pour qu'on arrive à la fin.

AmauryCarrade commented:
J'approuve et apprécie tes chipotages ! Merci d'avoir pris le temps. Je les traiterai tantôt.

AmauryCarrade commented:
Je retrouve le défilement horizontal sur [ce tutoriel](https://beta.zestedesavoir.com/tutoriels/885/introduction-a-la-preuve-de-programmes-c-avec-frama-c-et-son-greffon-wp/) sur Firefox Nightly pour Android. Le problème n'existe pas sur la prod.

AmauryCarrade committed:
Typos dans la documentation (merci artragis)

AmauryCarrade committed:
Sémantique dans les galeries

AmauryCarrade commented:
Toutes les remarques d'@artragis ont été traitées.

Situphen commented:
> Je retrouve le défilement horizontal sur [ce tutoriel](https://beta.zestedesavoir.com/tutoriels/885/introduction-a-la-preuve-de-programmes-c-avec-frama-c-et-son-greffon-wp/) sur Firefox Nightly pour Android. Le problème n'existe pas sur la prod.

Selon mon investigation, le coupable est la liste de contenus dans « Ces contenus pourraient vous intéresser » (`.content-item-list`) auquel il faudrait rajouter un petit `max-width: 100%` par exemple.

Il y a aussi un soucis de dépassement dans le bas de page du site (`.page-footer`) avec la liste des pages statiques (`.links`) qui dépasse sur tablette (en cause l'ajout du lien « Adhérer à l'association » j'imagine) mais ce soucis est déjà présent en production.

AmauryCarrade committed:
Correction de problèmes de défilement horizontaux

- Les listes de blocs de contenu étaient trop large sur mobile.

AmauryCarrade commented:
> > Je retrouve le défilement horizontal sur [ce tutoriel](https://beta.zestedesavoir.com/tutoriels/885/introduction-a-la-preuve-de-programmes-c-avec-frama-c-et-son-greffon-wp/) sur Firefox Nightly pour Android. Le problème n'existe pas sur la prod.
> 
> Selon mon investigation, le coupable est la liste de contenus dans « Ces contenus pourraient vous intéresser » (`.content-item-list`) auquel il faudrait rajouter un petit `max-width: 100%` par exemple.

Arf, j'avais déjà investigué et corrigeait quand tu as écrit ça, comme en témoigne le commit, mais merci !

> Il y a aussi un soucis de dépassement dans le bas de page du site (`.page-footer`) avec la liste des pages statiques (`.links`) qui dépasse sur tablette (en cause l'ajout du lien « Adhérer à l'association » j'imagine) mais ce soucis est déjà présent en production.

Ça par contre j'avais pas vu, à voir ce qu'il est possible de faire. Je regarde.

AmauryCarrade committed:
Défilements horizontaux : épisode 2, le pied

- Les liens du pied de page étaient trop large sur tablette. Ils ont
  été réduits en reformulant en fonction de la largeur de l'écran.

AmauryCarrade commented:
> Il y a aussi un soucis de dépassement dans le bas de page du site (`.page-footer`) avec la liste des pages statiques (`.links`) qui dépasse sur tablette (en cause l'ajout du lien « Adhérer à l'association » j'imagine) mais ce soucis est déjà présent en production.

@Situphen Résolu en reformulant le texte sur les petits écrans. Merci ! N'hésitez pas si vous avez une opposition à écrire le texte comme ça.

![Le lien « Adhérer à l'association » devient simplement « adhérer » sur tablette et inférieur.](https://user-images.githubusercontent.com/1417570/103235800-4093da00-4943-11eb-9a3f-c0a4af3a366a.png)

philippemilink commented:
Bêta à jour.

Situphen commented:
> Je retrouve le défilement horizontal sur [ce tutoriel](https://beta.zestedesavoir.com/tutoriels/885/introduction-a-la-preuve-de-programmes-c-avec-frama-c-et-son-greffon-wp/) sur Firefox Nightly pour Android. Le problème n'existe pas sur la prod.

Ce soucis est toujours présent sur tablette. :)

AmauryCarrade committed:
Défilements horizontaux : épisode 3, le retour des cartes

AmauryCarrade commented:
> > Je retrouve le défilement horizontal sur [ce tutoriel](https://beta.zestedesavoir.com/tutoriels/885/introduction-a-la-preuve-de-programmes-c-avec-frama-c-et-son-greffon-wp/) sur Firefox Nightly pour Android. Le problème n'existe pas sur la prod.
> 
> Ce soucis est toujours présent sur tablette. :)

Mais de quoi parles-tu

Situphen commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,254 @@
+
+Les styles de Zeste de Savoir sont complexes, formés de l'empilement d'années
+de contributions par des personnes très différentes. Pour qu'ils restent
+maintenable, ce document propose une norme et des recommandations.
```
Accord avec "ils" ?
```suggestion
maintenables, ce document propose une norme et des recommandations.
```

Situphen commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,254 @@
+
+Afin de garantir une cohérence à l'interface de Zeste de Savoir, plusieurs
+éléments graphiques sont normés. On ne parle pas ici de composants graphiques
+tel des boutons, des dialogues…, mais plutôt des éléments graphiques qui les
```
Accord avec "composants graphiques" ?
```suggestion
tels des boutons, des dialogues…, mais plutôt des éléments graphiques qui les
```

AmauryCarrade replied:
Il ne me semblait pas qu'on accordait ici, mais tu sembles avoir raison.

Situphen commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,254 @@
+composent : couleurs, polices, tailles, ombres…
+
+Tout développement graphique sur Zeste de Savoir **doit** respecter cette norme
+(avec quelque exceptions mentionnées ci-après).
```
Accord avec "exceptions" ?
```suggestion
(avec quelques exceptions mentionnées ci-après).
```

Situphen commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,254 @@
+   *voilà*. Aussi, ça évite de *deviner* quelle valeur choisir au hasard.
+
+   On peut enfin ajouter un troisième argument : ces normes ont été étudiées
+   dans une optique d'accessbilité — à supposer qu'elles soient correctement
```
```suggestion
   dans une optique d’accessibilité — à supposer qu'elles soient correctement
```

Situphen commented on doc/source/front-end/charte-interface.rst:
```diff
@@ -0,0 +1,254 @@
+
+.. attention::
+   Cette norme peut être ignorée s'il s'agit d'aligner des éléments au pixel près. Cela dit, un code
+   CSS moderne ne devrait pas nécessité ce genre de choses. Par contre, certains éléments plus anciens
```
```suggestion
   CSS moderne ne devrait pas nécessiter ce genre de choses. Par contre, certains éléments plus anciens
```

AmauryCarrade replied:
Argh

Situphen reviewed (changes requested):
Quelques petites coquilles dans la documentation

AmauryCarrade committed:
Typos in documentation (thanks Situphen)

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

AmauryCarrade committed:
La bannière globale masquait le dernier élément du menu

AmauryCarrade committed:
$size-* renommé en $font-size-*

Plus clair et évite des conflits lorsque la norme est utilisée dans
d'autres projets (par exemple avec Bulma).

AmauryCarrade committed:
Mise à jour des normes dans la documentation

AmauryCarrade committed:
Mise à jour des liens dans les normes

AmauryCarrade committed:
Alignement correct des titres de l'accueil.

- Ils sont reconstruits avec flex.
- À bas les flottants.
- Qu'ils coûlent.

AmauryCarrade committed:
Alignement correct des titres avec boutons.

- Ils sont reconstruits avec flex, eux aussi.
- À bas les flottants, encore et encore.
- Qu'ils coûlent, d'accord, d'accord.

AmauryCarrade committed:
Manque de marges entre sections de liste de contenus

AmauryCarrade committed:
Correction de l'alignement des icônes des titres

AmauryCarrade committed:
Manque de marges etre sections de liste de contenus

Pour de vrai

Situphen commented:
Deux approbations, de très nombreuses corrections (y compris sur des détails), et un rendu très propre : c'est l'heure de la fusion ! Il restera forcément quelques coquilles par-ci par-là mais rien de très grave car elles sont mineures et seront probablement corrigées avant la mise en production de la prochaine version.

Merci @AmauryCarrade pour ce travail d'enfer !

Je garde les commits séparés car même s'ils ne sont pas tous parfaitement atomiques ils sont bien nommés et puis en un seul commit ça ferait beaucoup trop.
